src/helpers/NodeHelper.ts

```diff
@@ -0,0 +1,113 @@
+import { StaticJsonRpcProvider } from "@ethersproject/providers";
+import { minutesAgo } from "./index";
+import { EnvHelper } from "./Environment";
+
+interface ICurrentStats {
+  failedConnectionCount: number;
+  lastFailedConnectionAt: number;
+}
+
+interface IInvalidNode {
+  key: string;
+  value: number;
+}
+
+/**
+ * NodeHelper used to parse which nodes are valid / invalid, working / not working
+ * NodeHelper.currentRemovedNodes is Object representing invalidNodes
+ * NodeHelper.logBadConnectionWithTimer logs connection stats for Nodes
+ * NodeHelper.getNodesUris returns an array of valid node uris
+ */
+export class NodeHelper {
```

Comment:
all the node failure / fallback is handled in this class

Reply:
the only two publicly used functions in this class right now are:

1. `NodeHelper.getNodesURIs` which is the same as our previous ALL_URIS but filters out `invalidNodes`
2. `NodeHelper.logBadConnectionWithTimer` which logs when a connection fails. used within try/catch of async node requests.